dexlib2/src/main/java/org/jf/dexlib2/writer/DexWriter.java

```diff
@@ -166,6 +170,22 @@ protected DexWriter(Opcodes opcodes) {
         this.annotationSection = sectionProvider.getAnnotationSection();
         this.annotationSetSection = sectionProvider.getAnnotationSetSection();
         this.encodedArraySection = sectionProvider.getEncodedArraySection();
+
+        overflowableSections = new IndexSection<?>[] {
+                //stringSection,            // supports jumbo indexes
+                typeSection,
+                //protoSection,             // reduntant check: cannot be larger than methodSection
+                fieldSection,
+                methodSection,
+                //classSection,             // reduntant check: cannot be larger than typeSection
+                callSiteSection,
+                methodHandleSection,
+
+                //typeListSection,          // is not an indexed section
```

Comment:
Why even mention these here?

Reply:
there are i think 3 places in code in which the list of sections is repeated. i unified the look of the 3 lists and made a comment that reminds maintainers of updating the other 2 lists whenever they update the main field definitions. (this would have help you last time when you forgot doing exactly that.) i thought that the full list with explicit rationales for exclusions was better than magically missing sections, so that's what i chose.

this way anyone looking at the code does not need to wonder whether missing sections are by omission o decision.

Reply:
Given that this is an array of IndexedSections, it doesn't make any sense to include or call out non-indexed sections, imo.

Reply:
see your point, i'll take it out. just give me some time to fix things as i'm tied up right now. thanks!

Reply:
Given my tardiness in reviewing it, I can't complain :)

Reply:
removed comments referencing non-indexed sections